frontend/javascripts/dashboard/advanced_dataset/create_explorative_modal.tsx

```diff
@@ -63,6 +70,82 @@ export function NewVolumeLayerSelection({
   );
 }
 
+export function RestrictResolutionSlider({
+  datasetResolutionInfo,
+  selectedSegmentationLayer,
+  resolutionIndices,
+  setResolutionIndices,
+}: RestrictResolutionSliderProps) {
+  let highestResolutionIndex = datasetResolutionInfo.getHighestResolutionIndex();
+  let lowestResolutionIndex = datasetResolutionInfo.getClosestExistingIndex(0);
+
+  if (selectedSegmentationLayer != null) {
+    const datasetFallbackLayerResolutionInfo = getResolutionInfo(
+      selectedSegmentationLayer.resolutions,
+    );
+    highestResolutionIndex = datasetFallbackLayerResolutionInfo.getHighestResolutionIndex();
+    lowestResolutionIndex = datasetFallbackLayerResolutionInfo.getClosestExistingIndex(0);
+  }
+
+  const highResolutionIndex = Math.min(highestResolutionIndex, resolutionIndices[1]);
+  const lowResolutionIndex = Math.max(lowestResolutionIndex, resolutionIndices[0]);
+
```

Comment:
See my other comment. This component should ideally ensure that the low/high values are always valid. When changing these values, this is already ensured by antd since you provide the correct min/max limits. However, if the user never changes the slider, the initial values are likely invalid.

Therefore, I suggest to set the values correctly on initial render (or in case the valid range changes which probably can never happen). This can be done with `useEffect` which will execute the passed function, if one of the elements of the second parameter (the "dependencies") change. Adding the following after line 91 should do the trick:

```
  useEffect(() => {
    setResolutionIndices([lowestResolutionIndex, highestResolutionIndex]);
  }, [lowestResolutionIndex, highestResolutionIndex]);
```

(`useEffect` needs to be imported)